### PR TITLE
[node-manager] caps: allow to skip bootstrap phases for nodes

### LINKED ISSUE #8494: [node-manager]  Cluster API Provider Static and master node-group
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

When bootstraping a cluster with CAPS and needing 3 master nodes
I boot one of them with `dhctl boostrap`, for the rest of the master nodes I need to create a additional master `node group` and `StaticInstance`.
This solution seems inconsistent. It turns out that I will have a StaticInstance for two masters and no StaticInstance for the third master.
In our quickstart, we suggest using StaticInstance resources and there is no mention of `manual-bootstrap-for-master`. We are switching from these scripts to StaticInstance, and commander doesn't know how to work with these scripts.

### Proposed Solution

we have hack, but count = 2 is not 3 master nodes in cluster

```yaml
---
apiVersion: deckhouse.io/v1
kind: NodeGroup
metadata:
  name: master
spec:
  disruptions:
    approvalMode: Manual
  nodeTemplate:
    labels:
      node-role.kubernetes.io/control-plane: ""
      node-role.kubernetes.io/master: ""
    taints:
    - effect: NoSchedule
      key: node-role.kubernetes.io/control-plane
  nodeType: Static
  staticInstances:
    count: 2   # <------------------------------ 
    labelSelector:
      matchLabels:
        type: master
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by mvasl**
## Description

Added support of adopting manually bootstrapped nodes under control of CAPS via StaticInstance's `static.node.deckhouse.io/skip-bootstrap-phase: ""` annotation.

Closes #8494

## Why do we need it, and what problem does it solve?

We want CAPS to manage our first master node.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: feature
summary: Cluster API Provider Static can adopt manually bootstrapped static nodes.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/static'...

**Review by RomanenkoDenys — APPROVED** [review]

File: modules/040-node-manager/images/caps-controller-manager/src/internal/controller/deckhouse.io/staticinstance_controller.go [hunk 1]
```diff
@@ -181,6 +189,71 @@ func (r *StaticInstanceReconciler) reconcileNormal(
...
+		}
+		if len(m.Status.Addresses) > 0 {
+			continue
+		}
+		if m.Status.FailureMessage != nil || m.Status.FailureReason != nil {
+			continue
+		}
+
+		tmp := m
+		machine = &tmp
```
[the remark this answers is not in the document]
Comment (mvasl) [reply modules/040-node-manager/images/caps-controller-manager/src/internal/controller/deckhouse.io/staticinstance_controller.go]: Done

[the remark this answers is not in the document]
Comment (RomanenkoDenys) [reply modules/040-node-manager/images/caps-controller-manager/src/internal/controller/deckhouse.io/staticinstance_controller.go]: You know what to do )

**Review by name212 — APPROVED** [review]

**Comment by name212** [comment]
@mvasl Can we add `static.node.deckhouse.io/skip-bootstrap-phase` for master node in the Static test?

**Review by name212 — CHANGES_REQUESTED** [review]
question

**Comment by mvasl** [comment]
We may try that once we fix daily e2e static

**Review by artbin — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

**Comment by mvasl** [comment]
/e2e/abort vclouddirector;Standard;containerd;1.27 9940323137 9940323137-1-con-1-27 /sys/deckhouse-oss/install:pr8753

**Pull request merged**